crates/ra_hir/src/expr.rs

```diff
@@ -516,23 +519,23 @@ impl ExprCollector {
         self.exprs.alloc(block)
     }
 
-    fn collect_expr(&mut self, expr: &ast::Expr) -> ExprId {
+    fn collect_expr(&mut self, expr: &ast::Expr, db: &impl HirDatabase) -> ExprId {
```

Comment:
I think we should add `DB: HirDatabase` field to `ExprCollector`, so as to avoid threading the arg explicitelly everywhere